packages/ui-concerto/src/lib/components/concertoForm.js

```diff
@@ -13,213 +13,156 @@
-    this.options = {
-      includeOptionalFields: true,
-      includeSampleData: 'sample',
-      disabled: props.readOnly,
-      visitor: new ReactFormVisitor(),
-      onFieldValueChange: (e, key) => {
-        this.onFieldValueChange(e, key);
-      },
-      addElement: (e, key, field) => {
-        this.addElement(e, key, field);
-      },
-      removeElement: (e, key, index) => {
-        this.removeElement(e, key, index);
-      },
-      ...props.options,
-    };
-    this.generator = new FormGenerator(this.options);
-  }
-
-  componentDidMount() {
-    this._loadAsyncData().then(modelProps => {
-      this.props.onModelChange(modelProps);
-    });
-  }
-
-  componentDidUpdate(prevProps) {
-    if (!isEqual(this.props.models, prevProps.models)) {
-      this._loadAsyncData().then(modelProps => {
-        this.props.onModelChange(modelProps);
-      });
-    }
-  }
-
-  static getDerivedStateFromProps(props) {
-    let shadowValue;
-    if (typeof props.json === 'string') {
-      try {
-        shadowValue = JSON.parse(props.json);
-      } catch {
-        // Do nothing
-      }
-    } else if (typeof props.json === 'object') {
-      shadowValue = { ...props.json };
-    }
-    return { value: shadowValue };
-  }
-
-  onFieldValueChange(e, key) {
-    const value = e.type === 'checkbox' ? e.checked : e.value;
-    // eslint-disable-next-line react/no-access-state-in-setstate
-    const valueClone = set({ ...this.state.value }, key, value);
-    this.setState({ value: valueClone });
-    this.props.onValueChange(valueClone);
-  }
-
-  async _loadAsyncData() {
-    this.setState({ loading: true });
-    const modelProps = await this.loadModelFiles(this.props.models, 'text');
-    this.setState({ loading: false });
-    return modelProps;
-  }
-
-  async loadModelFiles(files) {
-    let types;
-    let json;
-    let fqn = this.props.type;
-    try {
-      types = await this.generator.loadFromText(files);
-      // The model file was invalid
-    } catch (error) {
-      console.error(error.message);
-      // Set default values to avoid trying to render a bad model
-      // Don't change the JSON, it might be valid once the model file is fixed
-      return { types: [] };
-    }
-
-    if (types.length === 0) {
-      return { types: [] };
-    }
-
-    try {
-      if (
-        !types.map(t => t.getFullyQualifiedName()).includes(this.props.type)
-      ) {
-        fqn = types[0].getFullyQualifiedName();
-        json = this.generateJSON(fqn);
-        return { types, json };
-      }
-      json = this.generateJSON(this.props.type);
-    } catch (err) {
-      console.log(err);
-    }
-    return { types, json };
-  }
-
-  removeElement(e, key, index) {
-    const array = get(this.state.value, key);
+const ConcertoForm = (props) => {
+  const [value, setValue] = useState(typeof props.json === 'string' ? JSON.parse(props.json) : JSON.parse(JSON.stringify(props.json)));
```

Comment:
It seems quite odd that this `json` property accepts both JSON and a String. I wonder if there is a way to make this cleaner with the Storybook so we can remove this?

Reply:
If this was only a limitation of the storybook, we can only allow true objects here, since the storybook does not dictate the json anymore anyway.